### ISSUE TITLE
Bug: Rapidly changing teams can result in wrong categories being shown

### ISSUE THREAD

**Issue by chenilim** [mattermost/focalboard#2984]
## Steps to reproduce the behavior

1. Open Boards as a user who belongs to 2 or more teams
2. Create a category in each team, and (optionally) put a board in each
3. Rapidly click between the two teams
4. After a few seconds, the wrong category is shown for the team

## Expected behavior

The sidebar should always show the categories for the currently selected team.

## Screenshots (optional)

https://user-images.githubusercontent.com/46905241/165818998-de099733-d8f8-4d2d-8e04-08bc9182bec0.mov

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: v0.16 (bugbash2)
 - Browser and OS: FireFox on Mac

## Additional context (optional)

Extra observation: When the wrong category is shown, you are able to rename it, but it doesn't update until you switch back to the right team.

**Comment by sbishel** [comment]
Hey @chenilim - Could you retest this, I was unable to duplicate on the current `main`. Not sure if I wasn't clicking fast enough or if its been fixed.

I also attempted to duplicate [here](https://boards-seventwo2.test.mattermost.cloud/)

**Comment by chenilim** [comment]
I tried to repro this on the test server above with Chrome on Mac, but could not. However, it does kinda repro with FireFox on Mac:

https://user-images.githubusercontent.com/46905241/174688604-6f93d625-fc4e-42bb-8f6a-5ea05898aed8.mov

as well as with Safari:

https://user-images.githubusercontent.com/46905241/174689413-a3e6f584-f807-40f0-8c2f-0eeee169f195.mov

With 3 teams, I was able to get into a state where team 3 was selected, but the boards for team 2 appear. However, clicking on a board in the sidebar causes the UI to update to the correct state. So, I'd propose lowering this to sev-3, and leaving this open as a known issue. cc @wuwinson.
